superset/assets/src/dashboard/components/filterscope/FilterFieldTree.jsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import React from 'react';
+import PropTypes from 'prop-types';
+import CheckboxTree from 'react-checkbox-tree';
+import { t } from '@superset-ui/translation';
+
+import 'react-checkbox-tree/lib/react-checkbox-tree.css';
+import {
+  CheckboxChecked,
+  CheckboxUnchecked,
+  CheckboxHalfChecked,
+} from '../../../components/CheckboxIcons';
+import renderFilterFieldTreeNodes from './renderFilterFieldTreeNodes';
+import { filterScopeSelectorTreeNodePropShape } from '../../util/propShapes';
+
+const propTypes = {
+  activeKey: PropTypes.oneOfType([null, PropTypes.string]),
+  nodes: PropTypes.arrayOf(filterScopeSelectorTreeNodePropShape).isRequired,
+  checked: PropTypes.arrayOf(
```

Comment:
I remember your comment from the previous PR, what this really is is `checkedIds` or `checkedList`. `checked` by itself seems very much like a boolean argument. I know that's what the library uses, but can we update to one of the more explicit prop names to make this clearer?

I still don't understand why this needs to allow both numbers and strings too. Can we enforce one or the other coming in so that we don't have this multityped array?